src/java/voldemort/store/readonly/ReadOnlyStorageConfiguration.java

```diff
@@ -48,7 +49,18 @@ public ReadOnlyStorageConfiguration(VoldemortConfig config) {
         this.nodeId = config.getNodeId();
         this.deleteBackupMs = config.getReadOnlyDeleteBackupMs();
         this.maxValueBufferAllocationSize = config.getReadOnlyMaxValueBufferAllocationSize();
-        this.voldConfig = config;
+
+        FailedFetchLock failedFetchLockInstance;
+        if (config.getHighAvailabilityStateAutoCleanUp()) {
+            try {
+                failedFetchLockInstance = FailedFetchLock.getLock(config, null);
```

Comment:
My latest commit closes the ```FailedFetchLock``` and its underlying Hadoop FS as soon as it's finished with it. It also instantiates just one ```FailedFetchLock``` to deal with all stores in the offline to online transition, so the memory footprint should be minimal.